tests/e2e/utils/utils.go

```diff
@@ -0,0 +1,148 @@
+package utils
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+const (
+	DeleteNSTimeout = 10 * time.Minute
+
+	SingleCallTimeout = 5 * time.Minute
+
+	// How often to Poll pods, nodes and claims.
+	Poll = 2 * time.Second
+
+	pollShortTimeout = 1 * time.Minute
+	pollLongTimeout  = 5 * time.Minute
+
+	PodImage = "k8s.gcr.io/pause:3.1"
+)
+
+func findExistingKubeConfig() string {
+	defaultKubeConfig := "/etc/kubernetes/admin.conf"
+	// locations using DefaultClientConfigLoadingRules, but also consider `defaultKubeConfig`.
+	rules := clientcmd.NewDefaultClientConfigLoadingRules()
+	rules.Precedence = append(rules.Precedence, defaultKubeConfig)
+	return rules.GetDefaultFilename()
+}
+
+// GetClientSet obtains the client set interface from Kubeconfig
+func GetClientSet() (clientset.Interface, error) {
+	//TODO: It should implement only once
+	//rather than once per test
+	filename := findExistingKubeConfig()
+	//fmt.Printf(filename)
+	c := clientcmd.GetConfigFromFileOrDie(filename)
+	restConfig, err := clientcmd.NewDefaultClientConfig(*c, &clientcmd.ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: ""}}).ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	clientSet, err := clientset.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+	return clientSet, nil
+}
+
+// ExtractDNSPrefix obtains the cluster DNS prefix
+func ExtractDNSPrefix() string {
+	c := ObtainConfig()
+	return c.CurrentContext
+}
+
+// ExtractRegion obtains the cluster region
+func ExtractRegion() string {
+	c := ObtainConfig()
+	prefix := ExtractDNSPrefix()
+	url := c.Clusters[prefix].Server
+	domain := strings.Split(url, ".")
```

Comment:
can just return '{location}.cloudapp.azure.com' as a whole. For now master and agents would be in same region and share same endpoint